devices/Button/ButtonBase.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Threading;
+
+namespace Iot.Device.Button
+{
+    /// <summary>
+    /// Base implementation of Button logic.
+    /// Hardware independent. Inherit for specific hardware handling.
+    /// </summary>
+    public class ButtonBase : IDisposable
+    {
+        internal const int DEFAULT_DOUBLE_PRESS_TICKS = 15000000;
```

Comment:
Please use naming which are not C style but C# style. Only exceptions are for registers. In this case, it will be:
```suggestion
        internal const int DefaultDoublePressTicks = 15000000;
```